### PR TITLE
APPS-2530 prerequisits for user consent managment

### PR DESCRIPTION
**Pull request by TimurSultanov**
## PR Description

Ticket: https://spryker.atlassian.net/browse/APPS-2530

### CODE REVIEW

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: Prerequisites for adding user consent management
+description: Learn about the steps you need to take before you can start working with consent managers through AOP.
+template: howto-guide-template
+---
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:5]: is this table required?

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This document describes the prerequisites for using consent managers through AOP
+
+### Required modules:
+
+* spryker/cms-slot
+* spryker/asset-external
+* spryker/asset-external-storage
+* spryker-shop/asset-external-widget
+* spryker/event
+* spryker/event-aws-sns-broker (?)
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:16]: mark both with  with the question mark.

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### Event listeners:
+
+* AssetExternalStorageEventSubscriber (`\Spryker\Zed\AssetExternalStorage\Communication\Plugin\Event\Subscriber\AssetExternalStorageEventSubscriber`)
+* AssetExternalSubscriber (`\Spryker\Zed\AssetExternal\Communication\Plugin\Event\Subscriber\AssetExternalSubscriber`)
+
+### Rabbit MQ queues:
+
+* sync.storage.cms
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:25]: Code snippet is needed, how to enabled it

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+### CMS Slots:
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:27]: please add a link to CMS slots overview in Spryker docs.

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Example of a data import line:
+```csv
+@ShopUi/templates/page-blank/page-blank.twig,external-asset-header-top,SprykerAssetExternal,"External header top","In the very top position in the header of all pages",1 `
+```
+
+### CMS Slot content plugins:
+
+* SprykerAssetExternal: \SprykerShop\Yves\AssetExternalWidget\Plugin\ShopCmsSlot\AssetExternalWidgetCmsSlotContentPlugin
+
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:39]: code snippet is needed

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+* AssetExternalStorageEventSubscriber (`\Spryker\Zed\AssetExternalStorage\Communication\Plugin\Event\Subscriber\AssetExternalStorageEventSubscriber`)
+* AssetExternalSubscriber (`\Spryker\Zed\AssetExternal\Communication\Plugin\Event\Subscriber\AssetExternalSubscriber`)
+
+### Rabbit MQ queues:
+
+* sync.storage.cms
+
+### CMS Slots:
+
+* external-asset-header-top
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:29]: is it hadrcoded  in the code? or provided in event?

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+Example of a data import line:
+```csv
+@ShopUi/templates/page-blank/page-blank.twig,external-asset-header-top,SprykerAssetExternal,"External header top","In the very top position in the header of all pages",1 `
+```
```
Comment (matweew) [inline docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:34]: please also add a console command to import it

**Review by matweew — COMMENTED** [review]
looking into this document it should be clear full scope WHAT should be done to have these prerequisites.

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md (hunk 1 above)
Comment (TimurSultanov) [reply docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:5]: as I understand, yes

File: docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md (hunk 6 above)
Comment (TimurSultanov) [reply docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:29]: app-store-suite sends an event with this CMS slot, it should be added on a project level beforehand via data import

Comment (matweew) [reply docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:29]: ok is it hardcoded on PBC side? or user have to enter in manually in config form?

Comment (TimurSultanov) [reply docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:29]: it's hardcoded on PBC side

Comment (matweew) [reply docs/aop/dev/prerequisites-for-adding-apps/prerequisites-for-adding-user-consent-management.md:29]: ok

**Pull request closed without merging**